traits/trait_types.py

```diff
@@ -365,6 +366,30 @@ class Str ( BaseStr ):
     fast_validate = ( 11, basestring )
 
 
+class IdentifierStr(BaseStr):
+    #: A description of the type of value this trait accepts:
+    info_text = 'an ASCII identifier string'
+
+    #: The default value when first created
+    default_value = Undefined
+
+    # Regexp for identifier recognition
+    _identifier_regex = re.compile("[_A-Za-z][_a-zA-Z0-9]*$")
+
+    def validate ( self, object, name, value ):
+        """ Validates that a specified value is valid for this trait.
+        """
+
+        if (isinstance( value, str ) and
```

Comment:
I wouldn't add this to `traits`, then. Almost every time I have needed a Python identifier trait, it has been as part of a UI, which necessarily requires `unicode` objects. If your particular use case strictly requires a `str` object, I recommend keeping this trait definition as part of that application.